File: Client/GameBox.Admin.UI/Model/UserModel.cs
using System;
using System.Text.Json.Serialization;

namespace GameBox.Admin.UI.Model
{
    public class UserModel
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }

        [JsonPropertyName("username")]
        public string Username { get; set; }

        [JsonPropertyName("token")]
        public string Token { get; set; }

        [JsonPropertyName("expirationDate")]
        public DateTime ExpirationDate { get; set; } = DateTime.Now;

        [JsonPropertyName("isAdmin")]
        public bool IsAdmin { get; set; }

        [JsonPropertyName("message")]
        public string Message { get; set; }
    }
}

File: Client/game_box_mobile_ui/lib/models/user_model.dart
class UserModel {
  final String id;
  final String username;
  final String token;
  final DateTime expirationDate;
  final bool isAdmin;

  UserModel({
    required this.id,
    required this.username,
    required this.token,
    required this.expirationDate,
    required this.isAdmin,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'] as String,
      token: json['token'] as String,
      isAdmin: json['isAdmin'] as bool,
      username: json['username'] as String,
      expirationDate: DateTime.parse(json['expirationDate'] as String),
    );
  }
}
